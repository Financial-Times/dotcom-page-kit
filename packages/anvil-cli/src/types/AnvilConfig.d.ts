import { Plugin } from '@financial-times/dotcom-page-kit-pluggable'

export interface AnvilConfig {
  plugins: Plugin[]
  settings?: {
    build?: {
      entry?: any
      targets?: any
    }
  }
}
