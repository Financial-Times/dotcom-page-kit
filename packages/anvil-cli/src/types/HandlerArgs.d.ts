import { CliContext } from '../entities/CliContext'
import * as pluggable from '@financial-times/anvil-pluggable'

export interface HandlerArgs extends pluggable.HandlerArgs {
  cli: CliContext
}
