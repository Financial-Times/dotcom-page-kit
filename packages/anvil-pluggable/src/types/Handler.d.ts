import { Pluggable } from '../entities/Pluggable'

interface HandlerArgs {
  resource: any
  pluggable: Pluggable
  publish: (hook: string, resource: any) => any
  [key: string]: any
}

export interface Handler {
  (args: HandlerArgs): any
}
