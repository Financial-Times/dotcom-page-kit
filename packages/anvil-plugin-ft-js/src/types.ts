export interface Settings {
  jsxPragma?: string
  jsxPragmaFrag?: string
  envTargets?: any
}
