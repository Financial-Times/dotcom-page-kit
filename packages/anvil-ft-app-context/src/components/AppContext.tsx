import React from 'react'
import { TAppContext } from '../types'
import { APP_CONTEXT_ELEMENT_ID } from '../constants'

export interface TAppContextProps {
  data: TAppContext
}

export function AppContext({ data }: TAppContextProps) {
  return (
    <script
      type="application/json"
      id={APP_CONTEXT_ELEMENT_ID}
      dangerouslySetInnerHTML={{ __html: JSON.stringify(data) }}
    />
  )
}
