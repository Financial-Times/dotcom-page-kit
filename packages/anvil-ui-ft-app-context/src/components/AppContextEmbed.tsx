import React from 'react'
import { TAppContext } from '../types'
import { APP_CONTEXT_ELEMENT_ID } from '../../constants'

export interface TAppContextProps {
  context: TAppContext
}

export function AppContextEmbed({ context }: TAppContextProps) {
  return (
    <script
      type="application/json"
      id={APP_CONTEXT_ELEMENT_ID}
      dangerouslySetInnerHTML={{ __html: JSON.stringify(context) }}
    />
  )
}

// test
