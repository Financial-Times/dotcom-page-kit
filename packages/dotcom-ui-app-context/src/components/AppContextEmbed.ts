import { ClientEmbed } from '@financial-times/dotcom-ui-client-embed'
import { TAppContext } from '../types'
import { APP_CONTEXT_ELEMENT_ID } from '../constants'

export type TAppContextProps = {
  appContext?: TAppContext
}

export function AppContextEmbed({ appContext }: TAppContextProps) {
  return ClientEmbed({ id: APP_CONTEXT_ELEMENT_ID, data: appContext })
}
