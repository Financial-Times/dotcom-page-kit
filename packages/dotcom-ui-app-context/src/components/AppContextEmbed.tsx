import React from 'react'
import { TAppContext } from '../types'
import { APP_CONTEXT_ELEMENT_ID } from '../constants'

export type TAppContextProps = {
  appContext?: TAppContext
}

export function AppContextEmbed({ appContext }: TAppContextProps) {
  return (
    <script
      type="application/json"
      id={APP_CONTEXT_ELEMENT_ID}
      dangerouslySetInnerHTML={{ __html: JSON.stringify(appContext) }}
    />
  )
}
