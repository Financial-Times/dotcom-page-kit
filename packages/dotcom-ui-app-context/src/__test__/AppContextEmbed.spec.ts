import renderer from 'react-test-renderer'
import { AppContextEmbed as subject } from '../components/AppContextEmbed'
import { TAppContext } from '../types'

const fakeContext: TAppContext = {
  appName: 'app-name',
  appVersion: '123',
  edition: 'uk',
  product: 'next',
  abTestState: 'someCohort:on',
  isProduction: true
}

describe('dotcom-ui-app-context/src/components/AppContextEmbed', () => {
  it('renders a script element containing app context properties', () => {
    const tree = renderer.create(subject({ appContext: fakeContext }))
    expect(tree).toMatchSnapshot()
  })
})
