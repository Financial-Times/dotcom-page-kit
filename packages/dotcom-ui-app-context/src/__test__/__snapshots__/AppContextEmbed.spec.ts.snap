// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotcom-ui-app-context/src/components/AppContextEmbed renders a script element containing app context properties 1`] = `
<script
  dangerouslySetInnerHTML={
    {
      "__html": "{"appName":"app-name","appVersion":"123","edition":"uk","product":"next","abTestState":"someCohort:on","isProduction":true}",
    }
  }
  id="page-kit-app-context"
  type="application/json"
/>
`;
